### DISCUSSION TITLE
Committing `codelists.json` to the repo

### DISCUSSION THREAD

**Discussion by alexwalkerepi** [opensafely/documentation#170] [Codelists] [answered]
Perhaps for @inglesp:
It doesn't _seem_ to make a difference if I do or don't, but is it ever necessary or useful to commit the `codelists.json` to the repo? Is it intended as useful metadata for when the codelists were downloaded etc, or is it just used under the hood and not useful later? If the latter we could add it to the `.gitignore` in the research template.

**Comment by alexwalkerepi** [comment]
Ah, it seems it does make a difference to the github tests. It failed before I committed the changes to `codelists.json`:
https://github.com/opensafely/covid-vaccine-effectiveness-research/runs/2022648885?check_suite_focus=true

then passed after I committed the changes:
https://github.com/opensafely/covid-vaccine-effectiveness-research/runs/2022879219?check_suite_focus=true

**Comment by inglesp** [comment]
One for @evansd?

**Comment by evansd** [comment, marked as answer]
Yes, this should be committed alongside all the other files in the `codelists` directory.

At the moment you can get away with not including it at all because we wanted to avoid breaking existing studies so we put a workaround in place. The test output contains a warning about this but it gets lost in all the other output:
```
==> WARNING
    Using temporary workaround for Github Actions tests.
    You should run: opensafely codelists update
```

The reason we create this file is to make sure that the contents of the `codelists` directory exactly matches what was downloaded from OpenCodelists, but without having to contact OpenCodelists each time we run the tests which has the potential to make tests unreliable.
